db.go

```diff
@@ -735,33 +735,31 @@ func (d *DB) manualCompact(manual *manualCompaction) error {
 
 // Flush the memtable to stable storage.
 func (d *DB) Flush() error {
-	if atomic.LoadInt32(&d.closed) != 0 {
-		panic(ErrClosed)
-	}
-
-	d.mu.Lock()
-	mem := d.mu.mem.mutable
-	err := d.makeRoomForWrite(nil)
-	d.mu.Unlock()
+	flushDone, err := d.AsyncFlush()
 	if err != nil {
 		return err
 	}
-	<-mem.flushed()
+	<-flushDone
 	return nil
 }
 
 // AsyncFlush asynchronously flushes the memtable to stable storage.
-//
-// TODO(peter): untested
-func (d *DB) AsyncFlush() error {
+func (d *DB) AsyncFlush() (chan struct{}, error) {
```

Comment:
I unified `AsyncFlush()` and `Flush()` because I thought `AsyncFlush()` would be needed for my test case. While it is still used there, I do not think it ended up being required, so we can revert these changes if they're unappealing.